pkgs/development/python-modules/hatchling/default.nix

```diff
@@ -0,0 +1,58 @@
+# https://nixos.org/manual/nixpkgs/stable/#developing-with-python
+{ lib
+, buildPythonPackage
+, fetchPypi
+, intreehooks
+, pythonOlder
+, isPy27
+
+# All runtime dependencies
+, editables ? null
+, importlib-metadata
+, packaging
+, pathspec
+, pluggy
+, toml
+, tomli
+}:
+
+buildPythonPackage rec {
+  pname = "hatchling";
+  version = "0.19.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "942b34e454ce3be5b03139e0a198ba8ebc08a6095d345fe829b1ae92e9326c2c";
+  };
+
+  nativeBuildInputs = [
+    intreehooks
+  ];
+
+  buildInputs = [
+    packaging
+    pathspec
+    pluggy
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    importlib-metadata
+  ] ++ lib.optionals isPy27 [
+    toml
+  ] ++ lib.optionals !isPy27 [
+    tomli
+  ];
+  propagatedBuildInputs = buildInputs ++ [
+    editables
+  ];
```

Comment:
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies-handling-dependencies
no need to have python deps in buildInputs unless they're build time only